api/main.go

```diff
@@ -93,6 +122,75 @@ func getSegmentAndIds(w http.ResponseWriter, r *http.Request) {
 			fmt.Fprintf(w, "%s \n", b)
 		}
 	}
+
+}
+
+func getContactIdByEmail(w http.ResponseWriter, r *http.Request, contactEmail string) string {
+	// Mautic auth
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", mauticURL+"api/contacts?search=email:+"+contactEmail, nil)
+	req.SetBasicAuth(mauticUser, mauticPW)
+	resp, err := client.Do(req)
+
+	contactId := ""
+
+	// Get contact ID from response
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprintf(w, "Mautic HTTP request failed with error %s\n", err)
+		return contactId
+	}
+
+	bodyText, _ := ioutil.ReadAll(resp.Body)
+	dec := json.NewDecoder(strings.NewReader(string(bodyText)))
+	for {
+		var data ContactInfoByEmail
+		if err := dec.Decode(&data); err == io.EOF {
+			break
+		} else if err != nil {
+			fmt.Fprintf(w, "Decode failed with error %s\n", err)
+		}
+
+		// Get contact ID
+		for key := range data.Contacts {
+			contactId = key
+		}
+	}
+	return contactId
+}
+
+func getContactSegmentsById(w http.ResponseWriter, r *http.Request, contactId string) map[string]bool {
```

Comment:
this fn looks very similar to `getContactIdByEmail`, maybe consider making a more general one that handles both but just with a type difference?

Reply:
I'm not sure if this is possible since `getContactIdByEmail` simply gets the Contact ID, whereas `getContactSegmentsById` gets the Contact's Segment IDs and then marks these segments down on a map.